### ISSUE TITLE
Ability to change the shortcuts = keyboard UI

### ISSUE THREAD

**Issue by lazka** [quodlibet/quodlibet#535]
Original [issue 535](https://code.google.com/p/quodlibet/issues/detail?id=535) created by arne.babenhauserheide on 2010-07-07T10:45:28.000Z:

Hi, 

I want to use quodlibet for playing background and theme music in pen-and-paper roleplaying games via my OLPC. 

“Queue song with Q“ is already a great tool for that, but since the OLPC doesn’t have a right CTRL key, “ctrl-,” and “ctrl-.” are hard to reach. 

So I’d like to have an easy way to change the shortcuts. 

Using , and . without ctrl would be one example, but I’m sure that won’t be best for everyone. 

Also it would be nice to be able to use left-arrow and right-arrow in filesystem view to jump from the folder-list to the playlist and back (quickly selecting a new cathegory and playing a song from it). 

Is it possible to add keyboard mappings with plugins (for example to have something like keyboard themes for special use cases)? 

Best wishes and many thanks for writing the Quod Libet! 
Arne

**Comment by CreamyCookie** [comment]
@ArneBab Just out of curiousity, do you still use Quod Libet?

**Comment by ArneBab** [comment]
Yes - Quod Libet is the only audio player which really works well for me. It does what I need, and except for cases of broken gstreamer it just works. And keeps working.

From the usability side, the browser view is great, generating new playlists is fast, and I can play from disk. It can stop the music after some time (for going to bed) and wake me in the morning (that’s only when my wife and kids aren’t home ☺), and the different parts work well together. And it can cope with my 20k song library without a hitch. All in all Quod Libet is a joy to use.

**Comment by timpulver** [comment]
Being able to change shortcuts would be very helpful as there probably is no shortcut mapping which fits everyone.
A lot of programs use some sort of *shortcut recorder* for this.

**Comment by tukusejssirs** [comment]
I’ve just recently discovered QL (I needed a replacement for EasyTag, which lacked some stuff I needed), however, the inability to configure custom keybindings/shortcuts makes me less productive.

I wish that I could create/modify a keybinding for any QL action/command.

Anyway, I simply don’t understand why previous` song keybinding is <kbd>Ctrl</kbd>+<kbd>,</kbd> and next song keybinding <kbd>Ctrl</kbd>+<kbd>.</kbd>. Can’t they simply be <kbd>P</kbd> and <kbd>N</kbd> respectively?

Hoping this issue will be one of the top UX priority, seeing that many users wish to add/modify some particular keybindings; see #3229, #3223, #2452, #2256, #2038, #1930, #1884, #1774, #1701, #1651, #1288, #697, #563 (and probably #938).

**Comment by declension** [comment]
That's a good summary actually, thanks @tukusejssirs. I'm looking into a few of thos older tickets. Sadly though I have no real idea on how a global, configurable keybindings setup would work.

**Comment by tukusejssirs** [comment]
> Sadly though I have no real idea on how a global, configurable keybindings setup would work.

Well, I’m not a dev (I actually code a bit, but not in Python), therefore my input is somewhat unimportant.

Where exactly do you see the problem? In the algorithm/implementation? GUI? Elsewhere?

**Comment by scresante** [comment]
It would be nice if someone could just mention where in the source code these are defined. I would happy with hacking it myself to accomplish the customization I want, if I knew where to look.
